cocos/physics/framework/components/character-controllers/character-controller.ts

```diff
@@ -0,0 +1,521 @@
+    @tooltip('i18n:physics3d.character_controller.group')
+    public get group (): number {
+        return this._group;
+    }
+
+    public set group (v: number) {
+        if (DEBUG && !Number.isInteger(Math.log2(v >>> 0))) {
+            warn('[Physics]: The group should only have one bit.');
+        }
+        this._group = v;
+        if (this._cct) {
+            // The judgment is added here because the data exists in two places
+            if (this._cct.getGroup() !== v) this._cct.setGroup(v);
+        }
+    }
+
+    /**
+     * @en
+     * Gets or sets the minimum movement distance of the character controller.
+     * @zh
+     * 获取或设置角色控制器的最小移动距离。
+     */
+    @tooltip('i18n:physics3d.character_controller.minMoveDistance')
+    @type(CCFloat)
+    public get minMoveDistance () {
+        return this._minMoveDistance;
+    }
+
+    public set minMoveDistance (value) {
+        if (this._minMoveDistance === value) return;
+        this._minMoveDistance = Math.abs(value);
+    }
+
+    /**
+     * @en 、
+     * Gets or sets the stepOffset of the character controller.
+     * @zh
+     * 获取或设置角色控制器的最大爬台阶高度。
+     */
+    @tooltip('i18n:physics3d.character_controller.stepOffset')
+    @type(CCFloat)
+    public get stepOffset () {
+        return this._stepOffset;
+    }
+
+    public set stepOffset (value) {
+        if (this._stepOffset === value) return;
+        this._stepOffset = Math.abs(value);
+        if (this._cct) {
+            this._cct.setStepOffset(value);
+        }
+    }
+
+    /**
+     * @en
+     * Gets or sets the slope limit of the character controller in degree.
+     * @zh
+     * 获取或设置角色控制器的最大爬坡角度。
+    */
+    @tooltip('i18n:physics3d.character_controller.slopeLimit')
+    @type(CCFloat)
+    public get slopeLimit () {
+        return this._slopeLimit;
+    }
+
+    public set slopeLimit (value) {
+        if (this._slopeLimit === value) return;
+        this._slopeLimit = Math.abs(value);
+        if (this._cct) {
+            this._cct.setSlopeLimit(value);
+        }
+    }
+
+    /**
+     * @en
+     * Gets or sets the contact offset of the character controller.
+     * @zh
+     * 获取或设置角色控制器的接触间隙。
+     */
+    @tooltip('i18n:physics3d.character_controller.contactOffset')
+    @type(CCFloat)
+    public get contactOffset () {
+        return this._contactOffset;
+    }
+
+    public set contactOffset (value) {
+        if (this._contactOffset === value) return;
+        this._contactOffset = Math.abs(value);
+        if (this._cct) {
+            this._cct.setContactOffset(value);
+        }
+    }
+
+    // /**
+    //  * @en
+    //  * Gets or sets if the character controller can collide with other objects.
+    //  * @zh
+    //  * 获取或设置角色控制器是否和发生碰撞。
+    //  */
+    // @tooltip('i18n:physics3d.character_controller.detectCollisions')
+    // @type(CCBoolean)
+    // public get detectCollisions () {
+    //     return this._detectCollisions;
+    // }
+
+    // public set detectCollisions (value) {
```

Comment:
What's the default value of detect collision?

Reply:
![image](https://user-images.githubusercontent.com/9999378/236997612-cc02947b-2c3d-4384-914d-19f0471b356d.png)
![image](https://user-images.githubusercontent.com/9999378/236997690-ee5ab879-6fa4-4476-a9ac-0e1ffe07c73e.png)